src/modeler/AutoRest.Swagger/Validation/ProvidersPathValidation.cs

```diff
@@ -13,10 +13,10 @@
 
 namespace AutoRest.Swagger.Validation 
 {
-    public class ProviderPathsValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    public class ProvidersPathValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
     {
-        private readonly Regex FullRegex = new Regex(@"\/Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/[^\/]+(\/[^\/]+\/\{[^\}]+\})*(\/[\/^])?([\?]?[^\/]+)*$", RegexOptions.IgnoreCase);
-        private readonly Regex ProviderRegex = new Regex(@"Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/.+$", RegexOptions.IgnoreCase);
+        private Regex FullRegex = new Regex(@"\/Subscriptions\/\{.+\}\/ResourceGroups\/\{.+\}\/providers\/[^\/]+(\/[^\/]+\/\{[^\}]+\})*(\/[^\/]+)?([\?]?[^\/]+)*$", RegexOptions.IgnoreCase);
```

Comment:
I'm trying to catch the pattern that x-ms-paths follow (i.e. /path?foo={fooid})
I know it's a little overboard, but don't see the harm